rest/route/middleware_test.go

```diff
@@ -89,6 +89,42 @@ func TestPrefetchProject(t *testing.T) {
 	})
 }
 
+func TestNewProjectAdminMiddleware(t *testing.T) {
+	assert := assert.New(t)
+
+	ctx := context.Background()
+	opCtx := model.Context{}
+	opCtx.ProjectRef = &model.ProjectRef{
+		Private:    true,
+		Identifier: "orchard",
+		Owner:      "evergreen-ci",
+		Repo:       "evergreen",
+		Branch:     "master",
+		Admins:     []string{"johnny.appleseed"},
+	}
+
+	ctx = gimlet.AttachUser(ctx, &user.DBUser{Id: "giving.tree"})
+	r, err := http.NewRequest("GET", "/projects/orchard", nil)
+	assert.NoError(err)
+	assert.NotNil(r)
+
+	r = r.WithContext(context.WithValue(ctx, RequestContext, &opCtx))
+
+	mockConnector := &data.MockConnector{}
+	mw := NewProjectAdminMiddleware(mockConnector)
+	rw := httptest.NewRecorder()
+
+	mw.ServeHTTP(rw, r, func(rw http.ResponseWriter, r *http.Request) {})
+	assert.Equal(http.StatusUnauthorized, rw.Code)
+
+	ctx = gimlet.AttachUser(ctx, &user.DBUser{Id: "johnny.appleseed"})
```

Comment:
This test wouldn't catch the bug where attaching a user who isn't an admin is sufficient to auth to the admin middleware. Could you test with `gimlet.AttachUser(ctx, &user.DBUser{Id: "not.an.admin"})`?

Reply:
The first part of this test (line 106) covers that case, although is perhaps too tree-themed so I'll clarify.